### ISSUE TITLE
Question about the masked attention.

### ISSUE THREAD

**Issue by LiuShuaiyr** [facebookresearch/Mask2Former#3]
Where is the specific position of the formula (2) (page4  of paper) in the code ?
**X**l=softmax(**M**l-1+**Q**l**K**l)**V**l+**X**l-1

**Comment by bowenc0221** [comment]
PyTorch [torch.nn.MultiheadAttention](https://pytorch.org/docs/stable/generated/torch.nn.MultiheadAttention.html) already implements it. Thus we only need to provide `attn_mask` as input:https://github.com/facebookresearch/Mask2Former/blob/a03be8df3e583a2efbc1139d62b260bf22687259/mask2former/modeling/transformer_decoder/mask2former_transformer_decoder.py#L445

**Comment by LiuShuaiyr** [comment]
> PyTorch [torch.nn.MultiheadAttention](https://pytorch.org/docs/stable/generated/torch.nn.MultiheadAttention.html) already implements it. Thus we only need to provide `attn_mask` as input:
> 
> https://github.com/facebookresearch/Mask2Former/blob/a03be8df3e583a2efbc1139d62b260bf22687259/mask2former/modeling/transformer_decoder/mask2former_transformer_decoder.py#L445

Thank you for your reply.
